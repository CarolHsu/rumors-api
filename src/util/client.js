import config from 'config';
import elasticsearch from 'elasticsearch';

export default new elasticsearch.Client({
  host: config.get('ELASTICSEARCH_URL'),
  log: 'trace',
});
