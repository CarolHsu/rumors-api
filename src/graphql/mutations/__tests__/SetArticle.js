import GraphQL from 'util/GraphQL';
// import { loadFixtures, unloadFixtures } from 'util/fixtures';
import client from 'util/client';
// import fixtures from '../__fixtures__/SetArticle';

import MockDate from 'mockdate';

describe('SetArticle', () => {
  // beforeAll(() => loadFixtures(fixtures));

  it.only('creates articles', async () => {
    MockDate.set(1485593157011);
    const { data, errors } = await GraphQL({
      query: `mutation(
        $text: String!
        $references: [ArticleReferenceInput!]
      ) {
        SetArticle(
          text: $text
          references: $references
        ) {
          id
        }
      }`,
      variables: {
        text: 'FOO FOO', references: [{ type: 'LINE' }],
      },
    });
    MockDate.reset();

    expect(errors).toBeUndefined();

    console.log('data', data);

    const doc = await client.get({ index: 'articles', type: 'basic', id: data.SetArticle.id });
    delete doc._id;

    expect(doc).toMatchSnapshot();

    // Cleanup
    await client.delete({ index: 'articles', type: 'basic', id: data.SetArticle.id });
  });

  it('updates references for existing article, only 1 LINE reference is available, and permalink is filled for URL references');

  it('throws error if both id and text is not given', async () => {
    const { errors } = await GraphQL({
      query: `mutation(
        $references: [ArticleReferenceInput!]
      ) {
        SetArticle(
          references: $references
        ) {
          id
        }
      }`,
      variable: {
        references: [{ type: 'LINE' }],
      },
    });
    expect(errors).toMatchSnapshot();
  });

  // afterAll(() => unloadFixtures(fixtures));
});
