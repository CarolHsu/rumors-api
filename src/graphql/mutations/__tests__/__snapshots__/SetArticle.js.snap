exports[`SetArticle creates articles 1`] = `
Object {
  "_index": "articles",
  "_source": Object {
    "createdAt": "2017-01-28T08:45:57.011Z",
    "references": Array [
      Object {
        "type": "LINE",
      },
    ],
    "replyIds": Array [],
    "text": "FOO FOO",
    "updatedAt": "2017-01-28T08:45:57.011Z",
  },
  "_type": "basic",
  "_version": 1,
  "found": true,
}
`;

exports[`SetArticle throws error if both id and text is not given 1`] = `
Array [
  [GraphQLError: When id is not given (creating new Articles), text & references are required.],
]
`;
